biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+ */
+public class BlastXMLQuery implements ResultFactory {
+    private static final Logger log = Logger.getLogger(BlastXMLQuery.class.getName());
+    Document blastDoc = null;
+    private File targetFile;
+    private List<Sequence> queryReferences, databaseReferences;
+    private Map<String,Sequence> queryReferencesMap, databaseReferencesMap;
+    
+    @Deprecated
+    public BlastXMLQuery(String blastFile) throws Exception {
+        readFile(blastFile);
+    }
+    
+    public BlastXMLQuery() {
+        
+    }
+    public void setFile(File f){
+        targetFile = f;
+    }
+    /**
+     * 
+     * @param maxEScore
+     * @return
+     * @throws Exception 
+     */
+    @Deprecated
+    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
+        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();
+        log.info("Query for hits");
+        ArrayList<Element> elementList = XMLHelper.selectElements(blastDoc.getDocumentElement(), "BlastOutput_iterations/Iteration[Iteration_hits]");
+        log.info(elementList.size() + " hits");
+
+        for (Element element : elementList) {
+            Element iterationquerydefElement = XMLHelper.selectSingleElement(element, "Iteration_query-def");
+            String querydef = iterationquerydefElement.getTextContent();
+            Element iterationHitsElement = XMLHelper.selectSingleElement(element, "Iteration_hits");
+            ArrayList<Element> hitList = XMLHelper.selectElements(iterationHitsElement, "Hit");
+            for (Element hitElement : hitList) {
+                Element hitaccessionElement = XMLHelper.selectSingleElement(hitElement, "Hit_accession");
+                String hitaccession = hitaccessionElement.getTextContent();
+                Element hithspsElement = XMLHelper.selectSingleElement(hitElement, "Hit_hsps");
+                ArrayList<Element> hspList = XMLHelper.selectElements(hithspsElement, "Hsp");
+                for (Element hspElement : hspList) {
+                    Element evalueElement = XMLHelper.selectSingleElement(hspElement, "Hsp_evalue");
+                    String value = evalueElement.getTextContent();
+                    double evalue = Double.parseDouble(value);
+                    if (evalue <= maxEScore) {
+                        ArrayList<String> hits = hitsHashMap.get(querydef);
+                        if (hits == null) {
+                            hits = new ArrayList<String>();
+                            hitsHashMap.put(querydef, hits);
+                        }
+                        hits.add(hitaccession);
+                    }
+                }
+            }
+        }
+
+        return hitsHashMap;
+    }
+    
+    private void readFile(String blastFile) throws Exception{
+        log.info("Start reading " + blastFile);
+        blastDoc = XMLHelper.loadXML(blastFile);
+        log.info("Read finished");
+    }
+    

```

Comment:
Alright now I see what you wanted to do. In that case, unfortunately you really need to catch the exception explicitly:
```java
try {
} catch (SAXException e) {
// do something
} catch (ParserConfigurationException e) {
// do the same thing again
}
```
If we were in Java 7 (we really need to upgrade soon!) we could do that much more elegantly:
```java
try {
} catch (SAXException|ParserConfigurationException e) {
// do something
}
```
If you don't do this kind of explicit catching then you are catching any other exception that occurs within that try block (anywhere in the call hierarchy within the block), e.g. NullPointer, OutOfBounds or any other thing. In general it is a better idea to let those go through because here you are only concerned on what to do when this particular XML parsing fails.